File: common/src/main/java/net/impleri/playerskills/integration/ftbteams/FTBTeamApi.java
package net.impleri.playerskills.integration.ftbteams;

import dev.ftb.mods.ftbteams.FTBTeamsAPI;
import dev.ftb.mods.ftbteams.data.Team;
import dev.ftb.mods.ftbteams.data.TeamRank;
import net.impleri.playerskills.server.TeamApi;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class FTBTeamApi extends TeamApi {
    public static void registerInstance() {
        TeamApi.setInstance(new FTBTeamApi());
    }

    @Nullable
    private Team getTeam(UUID playerUuid) {
        return FTBTeamsAPI.getPlayerTeam(playerUuid);
    }

    protected List<UUID> getTeamMembersFor(UUID playerId) {
        var team = getTeam(playerId);

        if (team == null) {
            return new ArrayList<>();
        }

        return team.getRanked(TeamRank.NONE).keySet().stream().toList();
    }

}

File: common/src/main/scala/net/impleri/playerskills/integrations/ftbteams/FtbTeamApi.scala
package net.impleri.playerskills.integrations.ftbteams

import net.impleri.playerskills.integrations.ftbteams.facades.FtbTeamsApi
import net.impleri.playerskills.server.api.Team

import java.util.UUID

case class FtbTeamApi(ftbTeamApi: FtbTeamsApi = FtbTeamsApi()) extends Team {
  override protected[playerskills] def getTeamMembersFor(player: UUID): List[UUID] = {
    ftbTeamApi.getTeamOf(player)
      .toList
      .flatMap(_.getAllMembers)
  }
}
